fitlins/interfaces/nistats.py

```diff
@@ -149,31 +174,29 @@ def _run_interface(self, runtime):
         smoothing_fwhm = self.inputs.smoothing_fwhm
         if not isdefined(smoothing_fwhm):
             smoothing_fwhm = None
-
-        flm = level1.FirstLevelModel(
-            minimize_memory=False,
-            mask_img=mask_file, smoothing_fwhm=smoothing_fwhm)
-        flm.fit(img, design_matrices=mat)
+        if isinstance(img, nb.Cifti2Image):
+            labels, estimates = level1.run_glm(img.get_fdata(), mat.values)
+        else:
+            flm = level1.FirstLevelModel(
+                minimize_memory=False,
+                mask_img=mask_file, smoothing_fwhm=smoothing_fwhm)
+            flm.fit(img, design_matrices=mat)
 
         out_ents = self.inputs.contrast_info[0]['entities']
-        fname_fmt = os.path.join(runtime.cwd, '{}_{}.nii.gz').format
 
         # Save model level images
-        model_attr = {
-            'r_square': flm.r_square[0],
-            'log_likelihood': _get_voxelwise_logL(flm)[0]
-        }
-
-        model_maps = []
-        model_metadata = []
-        for attr, img in model_attr.items():
-            model_metadata.append(
-                {'stat': attr,
-                 **out_ents}
-                )
-            fname = fname_fmt('model', attr)
-            img.to_filename(fname)
-            model_maps.append(fname)
+        # model_attr = {
+        #     'r_square': flm.r_square[0],
+        #     'log_likelihood': _get_voxelwise_logL(flm)[0]
+        # }
+
+        # model_maps = []
+        # model_metadata = []
+        # for attr, img in model_attr.items():
+        #     model_metadata.append{'stat': attr, **out_ents})
+        #     fname = fname_fmt('model', attr)
+        #     img.to_filename(fname)
+        #     model_maps.append(fname)
```

Comment:
@adelavega @jdkent I had to filter this out for the moment because CIFTI-2s can't be fit with a `FirstLevelModel`. If either of you has some time to look at how to restore these with just the `run_glm` outputs, that would be great. I'll come back around to this after I get plotting working.